### PR TITLE
CI BDD tests with --no-docker mode

### PR DESCRIPTION
**Pull request by pietrzakacper**
Parametrizing BDD tests so it's possible to run them in NO_DOCKER mode.
One sequence required adding one sleep bc in --no-docker mode it was ending too fast.

### CODE REVIEW

**Review by MichalCz — APPROVED** [review]
Noice. :)

**Review by alicja-gruzdz — APPROVED** [review]

File: .github/workflows/sth-build-test.yml [hunk 1]
```diff
@@ -326,3 +326,122 @@ jobs:
       - name: Run BDD tests
         run: SCRAMJET_TEST_LOG=1 yarn test:bdd-ci
 
+  test-bdd-ci-no-host-no-docker-sth:
+    name: Test bdd-ci-no-host-no-docker STH (Nodejs ${{ matrix.node-version }})
+    needs: [analyze-code, build-sth, build-refapps, build-docker-sth-image, build-docker-runner-image, build-docker-pre-runner-image]
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:331]: So, in this case we don't need `build-docker-runner-image` and `build-docker-pre-runner-image`

```suggestion
    needs: [analyze-code, build-sth, build-refapps, build-docker-sth-image]
```

File: .github/workflows/sth-build-test.yml [hunk 2]
```diff
@@ -326,3 +326,122 @@ jobs:
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Run BDD for HUB params
+        run: NO_DOCKER=1 SCRAMJET_TEST_LOG=1 yarn test:bdd-ci-no-host
+
+
+  test-bdd-ci-sth-no-docker:
+      name: Test bdd-ci-no-docker STH (Nodejs ${{ matrix.node-version }})
+      needs: [analyze-code, build-sth, build-refapps, build-docker-sth-image, build-docker-runner-image, build-docker-pre-runner-image]
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:391]:
```suggestion
      needs: [analyze-code, build-sth, build-refapps, build-docker-sth-image]
```

File: .github/workflows/sth-build-test.yml [hunk 3]
```diff
@@ -326,3 +326,122 @@ jobs:
...
+
+      - name: Download dockerImg artifact 
+        uses: actions/download-artifact@v2
+        with:
+          name: dockerRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+      - name: Download dockerPreRunnerImg artifact 
+        uses: actions/download-artifact@v2
+        with:
+          name: dockerPreRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:369]: unnecessary
```suggestion
```

File: .github/workflows/sth-build-test.yml [hunk 4]
```diff
@@ -326,3 +326,122 @@ jobs:
...
+
+        - name: Download dockerImg artifact 
+          uses: actions/download-artifact@v2
+          with:
+            name: dockerRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+        - name: Download dockerPreRunnerImg artifact 
+          uses: actions/download-artifact@v2
+          with:
+            name: dockerPreRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:429]: unnecessary
```suggestion
```

Comment (daro1337) [inline .github/workflows/sth-build-test.yml:429]: unnecessary
```suggestion
```

File: .github/workflows/sth-build-test.yml [hunk 5]
```diff
@@ -326,3 +326,122 @@ jobs:
...
+          name: dockerPreRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+      - name: Unzip dockerImg-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+        run: pigz -d docker*Img*.tar.gz
+
+      - name: Unzip dist-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+        run: ls dist*tar.gz |xargs -n1 tar -I pigz -xf
+
+      - name: Load Docker images
+        run: ls -1  docker*Img*.tar| while read line; do docker load -i $line; done
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:368]: `docker load -i dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar` 
```suggestion
        run: docker load -i dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar
```

File: .github/workflows/sth-build-test.yml (hunk 1 above)
Comment (pietrzakacper) [reply .github/workflows/sth-build-test.yml]: fixed

File: .github/workflows/sth-build-test.yml (hunk 2 above)
Comment (pietrzakacper) [reply .github/workflows/sth-build-test.yml]: fixed

File: .github/workflows/sth-build-test.yml (hunk 3 above)
Comment (pietrzakacper) [reply .github/workflows/sth-build-test.yml]: fixed

File: .github/workflows/sth-build-test.yml (hunk 4 above)
Comment (pietrzakacper) [reply .github/workflows/sth-build-test.yml]: fixed

File: .github/workflows/sth-build-test.yml [hunk 6]
```diff
@@ -326,3 +326,122 @@ jobs:
...
+            name: dockerPreRunnerImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+        - name: Unzip dockerImg-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+          run: pigz -d docker*Img*.tar.gz
+
+        - name: Unzip dist-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+          run: ls dist*tar.gz |xargs -n1 tar -I pigz -xf
+
+        - name: Load Docker images
+          run: ls -1  docker*Img*.tar| while read line; do docker load -i $line; done
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:418]: `docker load -i dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar` 
```suggestion
          run: docker load -i dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar
```

File: .github/workflows/sth-build-test.yml [hunk 7]
```diff
@@ -326,3 +326,102 @@ jobs:
...
+        with:
+          name: dist-refapps-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz
+
+      - name: Download dockerSthImg artifact 
+        uses: actions/download-artifact@v2
+        with:
+          name: dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+      - name: Unzip dockerImg-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+        run: pigz -d docker*Img*.tar.gz
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:362]:
```suggestion
        run: pigz -d pigz -d dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
```

File: .github/workflows/sth-build-test.yml [hunk 8]
```diff
@@ -326,3 +326,102 @@ jobs:
...
+          with:
+            name: dist-refapps-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz
+
+        - name: Download dockerSthImg artifact 
+          uses: actions/download-artifact@v2
+          with:
+            name: dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
+
+        - name: Unzip dockerImg-${{ github.event.pull_request.head.sha }}-${{ matrix.node-version }}.tar.gz artifact
+          run: pigz -d docker*Img*.tar.gz
```
Comment (daro1337) [inline .github/workflows/sth-build-test.yml:412]:
```suggestion
          run: pigz -d pigz -d dockerSthImg-${{ matrix.node-version }}-${{ github.event.pull_request.head.sha }}.tar.gz
```

**Review by a-tylenda — APPROVED** [review]

**Pull request merged**